src/NUnitFramework/framework/Constraints/DelayedConstraint.cs

```diff
@@ -227,6 +227,7 @@ public override ConstraintResult ApplyTo<TActual>(TActual actual)
                 {
                     if (nextPoll > now)
                         ThreadUtility.BlockingDelay((int)TimestampDiff(delayEnd < nextPoll ? delayEnd : nextPoll, now).TotalMilliseconds);
+                    now = Stopwatch.GetTimestamp();
```

Comment:
This is correct, as otherwise *nextPoll* would be the same as when going into the method.
A nicer solution would be to only calculate *nextPoll* before the delay, removing two lines of code.
```csharp
while ((now = Stopwatch.GetTimestamp()) < delayEnd)
{
    long nextPoll = TimestampOffset(now, PollingInterval.AsTimeSpan);
    ThreadUtility.BlockingDelay((int)TimestampDiff(delayEnd < nextPoll ? delayEnd : nextPoll, now).TotalMilliseconds);

    ConstraintResult result = BaseConstraint.ApplyTo(actual);
    if (result.IsSuccess)
        return new DelegatingConstraintResult(this, result);
}
```

The test on `if (nextPoll > now)` is also not needed assuming `PollingInterval` is greater than 0.
The latter should be checked in the constraint constructor.
The constructor currently only tests `if (delayInMillisecond < 0)`, but should also not allow a negative *pollingIntervalInMilliseconds*.

Reply:
CAn you add this as a suggested commit?

Reply:
No, github doesn't let me suggest changes to non-changed lines.
I can push the change to your branch if you like.

Reply:
Please do :-)

Reply:
Change pushed. I have also added checks on pollInterval